massa-network-worker/src/network_worker.rs

```diff
@@ -96,55 +104,120 @@ impl NetworkWorker {
             controller_manager_rx,
         }: NetworkWorkerChannels,
         version: Version,
+        runtime: Runtime,
     ) -> NetworkWorker {
         let self_node_id = NodeId::new(keypair.get_public_key());
 
-        let (node_event_tx, node_event_rx) =
-            mpsc::channel::<NodeEvent>(cfg.node_event_channel_size);
+        let (conn_tx, connections_rx) =
+            bounded::<(ConnectionId, HandshakeReturnType)>(cfg.handshake_manager_channel_size);
+        let (handshake_tx, handshake_rx) =
+            bounded::<(ConnectionId, HandshakeWorker)>(cfg.handshake_manager_channel_size);
+        let handshake_join_handle =
+            start_handshake_manager(handshake_rx, conn_tx, runtime.handle().clone());
+
+        let (node_result_tx, node_result_rx) = bounded::<(
+            NodeId,
+            Result<ConnectionClosureReason, NetworkError>,
+        )>(cfg.node_result_channel_size);
+
+        let (handshake_peer_list_tx, handshake_peer_list_rx) =
+            bounded::<IpAddr>(cfg.handshake_peer_list_channel_size);
+
+        let (node_event_tx, node_event_rx) = bounded::<NodeEvent>(cfg.node_event_channel_size);
         let max_wait_event = cfg.max_send_wait_network_event.to_duration();
         NetworkWorker {
             cfg,
             self_node_id,
             keypair,
-            listener,
+            listener: Some(listener),
             establisher,
             peer_info_db,
             controller_command_rx,
             event: EventSender::new(controller_event_tx, node_event_tx, max_wait_event),
             controller_manager_rx,
             running_handshakes: HashSet::new(),
-            handshake_futures: FuturesUnordered::new(),
-            handshake_peer_list_futures: FuturesUnordered::new(),
+            handshake_peer_list_futures: Default::default(),
             node_event_rx,
             active_nodes: HashMap::new(),
-            node_worker_handles: FuturesUnordered::new(),
             active_connections: HashMap::new(),
             version,
+            runtime,
+            connections_rx,
+            handshake_tx,
+            handshake_join_handle,
+            node_result_rx,
+            node_result_tx,
+            handshake_peer_list_tx,
+            handshake_peer_list_rx,
         }
     }
 
     /// Runs the main loop of the network worker
     /// There is a `tokio::select!` inside the loop
```

Comment:
The tokio::select! comment seems outdated, it's the channel's select!() now.
Are the biases considerations the same between the two set of macros?